### ISSUE TITLE
Submit items that make it to the final step to Archive dot Org

### ISSUE THREAD

**Issue by AramZS** [PressForward/pressforward#994]
https://archive.readme.io/docs/creating-a-snapshot
https://www.w3.org/TR/annotation-model/

Proposed model:
```
{
  "@context": "http://www.w3.org/ns/anno.jsonld",
  "id": "http://example.org/anno2", <-- custom endpoint ?
  "type": "Annotation",
  "body": {
    "id": "http://example.org/analysis1.mp3", <-- Pressforward site
    "format": "audio/mpeg", <--- html
    "language": "fr" <--- Can we pull this from WP site settings?
  },
  "target": {
    "id": "http://example.gov/patent1.pdf", <--- item pulled to last step link
    "format": "application/pdf", <--- html
    "language": ["en", "ar"], <--- Again from WP site settings?
    "textDirection": "ltr", 
    "processingLanguage": "en"
  }
}
```

**Comment by yoannspace** [comment]
Question, you want archive dot org to save the page you are creating (the wordpress post) or to save the source page?

**Comment by AramZS** [comment]
The source page. The idea would be that we are adding the submitted page to our internal archive but also to Archive.org.

**Comment by yoannspace** [comment]
Great! We were thinking about the exact same thing and I realized it would make more sense to it in PF than in one of our other plugin.
Please don't wait for me, but I might be able to draft something up on this in the next weeks if you are interested. 2 questions: in which branch are you expecting this to be merged (5.2.x?)? when is 5.2 expected to be ready? Thanks

**Comment by AramZS** [comment]
Let's merge it in 5.2.x @yoannspace. I'm hoping that it will be ready to go in the next month or two. If not we may issue a patch release with this and smaller issues in it before a larger release to handle the big functionality we want to push.

**Comment by yoannspace** [comment]
Have you tried the API? I can't seem to be able to connect... https://archive.readme.io/docs/creating-a-snapshot
I successfully implemented a simple GET request to their server. I will add a PR here but not sure if it would be enough for you...

**Comment by boonebgorges** [comment]
This is a cool idea, with a few caveats:

- The mechanism at web.archive.org/save is extremely slow. I just ran some tests and it took 30+ seconds. We can't run it synchronously, at save time. We could do it in the browser, or on a cron job.
- Using their API https://archive.org/developers/tutorial-create-item.html is probably off the table, since it requires credentials.
- This feature involves pinging a third-party service. This is not quite the same as *tracking* users but it probably still falls under this guideline from wordpress.org: https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#7-plugins-may-not-track-users-without-their-consent The feature would have to have a toggle, and would probably have to be off by default.

**Comment by boonebgorges** [comment]
If we do this, we should save the fact that an item has been successfully added to archive.org. We can then show this to the user somehow, perhaps in our existing tools for showing PF metadata in the block editor. We could also use this as a potential fallback if a source item becomes unavailable; see #703.

**Comment by boonebgorges** [comment]
This is a fairly standalone feature, so I pulled it back into the 5.8.0 milestone.

The feature is now merged into the branch. It works like this:

- The feature is turned off by default, for reasons described above. It can be enabled at Dashboard > PressForward > Preferences > Site Settings
- When enabled, a post is submitted to archive.org when the post is promoted to Draft status (ie the final step in the PF workflow)
- archive.org submission is extremely slow, so the submission process is offloaded to a cron job
- Submission status (success, error, pending) is saved in the database
- When archive.org integration is enabled, and when you're editing a post that has a submission status, that status is shown in the PressForward section on the sidebar. I've built this only for the block editor, not for Classic. See screenshots.

![Screenshot_2024-12-11_13-25-49](https://github.com/user-attachments/assets/f7be87ba-7b60-4830-9410-fe8446dbbac2)
![Screenshot_2024-12-11_13-25-39](https://github.com/user-attachments/assets/dff76f1c-3ab5-48d8-8dfb-d01672b30fd4)
